django/contrib/admin/sites.py

```diff
@@ -409,6 +433,9 @@ def login(self, request, extra_context=None):
         request.current_app = self.name
         return LoginView.as_view(**defaults)(request)
 
+    def final_catch_all_view(self, request):
+        raise Http404
+
```

Comment:
This will _break_ the debug page I think: instead of a list of routed URL patterns, we'll get a plain 404 error response.  

(Would like to do something at the resolver level, but we don't have the request... Can we limit it just to `DEBUG=False`? 🤔)

Reply:
I'll look into solving this.

Reply:
Fixed this by including the tried URL patterns in the `technical_404_response` for raised `Http404`. If the URL resolves, it shows the patterns up unto the one used. I think this is a nice improvement regardless as it improves the technical 404 page outside the admin as well. As before, this only shows when `DEBUG` is `True`.

This also avoids the need to change runtime behavior based on `DEBUG` in a new way, which I think is also nice.

Reply:
OK, that sounds/looks interesting. Let me have a play. Thanks @jdufresne!